hazelcast/src/main/java/com/hazelcast/instance/NodeExtension.java

```diff
@@ -130,7 +131,7 @@
     /**
      * Creates factory method that creates server side client message handlers
      */
-    MessageTaskFactory createMessageTaskFactory();
+    List<? extends MessageTaskFactory> createMessageTaskFactories();
```

Comment:
We need list because different components can provide their own factories.
Yes ordering is relevant.